released/usecases/services_apps/sap_build_code/step2/main.tf

```diff
@@ -0,0 +1,269 @@
+# ------------------------------------------------------------------------------------------------------
+# Create the Cloud Foundry space
+# ------------------------------------------------------------------------------------------------------
+resource "cloudfoundry_space" "dev" {
+  name = "dev"
+  org  = var.cf_org_id
+}
+
+# ------------------------------------------------------------------------------------------------------
+# SETUP ALL SERVICES FOR CF USAGE
+# ------------------------------------------------------------------------------------------------------
+# ------------------------------------------------------------------------------------------------------
+# Setup sdm
+# ------------------------------------------------------------------------------------------------------
+# Entitle 
+resource "btp_subaccount_entitlement" "sdm" {
+  subaccount_id = var.subaccount_id
+  service_name  = "sdm"
+  plan_name     = "build-code"
+}
+
+# Create the service instance
+data "cloudfoundry_service" "sdm" {
+  name       = "sdm"
+  depends_on = [btp_subaccount_entitlement.sdm]
+}
+resource "cloudfoundry_service_instance" "sdm" {
+  name         = "default_sdm"
+  space        = cloudfoundry_space.dev.id
+  type         = "managed"
+  service_plan = data.cloudfoundry_service.sdm.service_plans["build-code"]
+  depends_on   = [cloudfoundry_space_role.space_manager, cloudfoundry_space_role.space_developer, cloudfoundry_org_role.organization_manager, btp_subaccount_entitlement.sdm]
+}
+# Create service key
+resource "random_id" "service_key_sdm" {
+  byte_length = 12
+}
+resource "cloudfoundry_service_credential_binding" "sdm" {
+  type             = "key"
+  name             = join("_", ["defaultKey", random_id.service_key_sdm.hex])
+  service_instance = cloudfoundry_service_instance.sdm.id
+}
+
+
+# ------------------------------------------------------------------------------------------------------
+# Setup mobile-services
+# ------------------------------------------------------------------------------------------------------
+# Entitle 
+resource "btp_subaccount_entitlement" "mobile_services" {
```

Comment:
Same generic remark as for step 1: would it make sense to make the creation of service instances and apps more flexible via boolean variables?

Reply:
Revised: To keep things self-contained, we leave the stuff in the CF step